### PR TITLE
#160845509 Add Search Functionality and Custom filtering based on parameters

### PR DESCRIPTION
**Pull request by reiosantos**
#### What does this PR do?
- Adds search functionality to articles

#### Description of Task to be completed?

- add Search Functionality and Custom filtering based on parameters
- Enable the user to filter articles using author name, article title or tag/keyword.

image

#### What are the relevant pivotal tracker stories?

[#160845509](https://www.pivotaltracker.com/story/show/160845509)

### CODE REVIEW

**Review by collinewait — DISMISSED** [review]

File: src/containers/viewAllArticles.js [hunk 1]
```diff
@@ -7,24 +7,28 @@ import ViewCard from '../components/articles/viewCard';
...
-			offset: 0
+			page: 1, limit: 20, offset: 0,
+			tag: '', title: '', author: '',
+			articles
 		};
 	}
 
 	componentWillMount() {
 		const { articles } = this.state;
+		const { results, count } = articles;
```
Comment (isheebo) [inline src/containers/viewAllArticles.js:27]: for this, we can have one destructuring assignment, that is
`const {results, count} = this.state.articles;`

File: src/containers/viewAllArticles.js [hunk 2]
```diff
@@ -7,24 +7,28 @@ import ViewCard from '../components/articles/viewCard';
...
 
-		if (!articles.results || articles.count < 1) {
+		if (!results || count < 1) {
```
Comment (isheebo) [inline src/containers/viewAllArticles.js:29]: is `results` a boolean? Please let's meet over this, if it is not

File: src/containers/viewAllArticles.js [hunk 3]
```diff
@@ -78,17 +85,44 @@ class ViewAllArticles extends React.Component {
...
+
+		return (
+			<div>
+				<div className='container'>
+					<div className={` ${styles['left-view']}`}>
+						<FilterArticles handleFilterSubmit={this.handleFilterSubmit} />
+					</div>
+
+					<div className={`col s12 l9 m9 ${styles['right-view']}`}>
+
```
Comment (isheebo) [inline src/containers/viewAllArticles.js]: let's meet over this too

File: src/reducers/articleReducers/articlesReducers.js [hunk 4]
```diff
@@ -43,11 +44,16 @@ const articlesReducerExtended = (state, action, allArticles) => {
 
 const articlesReducer = (state = articlesState, action) => {
 	const allArticles = state.results;
-
+	let { results = [] } = action.payload || {results: []};
```
Comment (isheebo) [inline src/reducers/articleReducers/articlesReducers.js]: this is too complex, my friend @reiosantos. Find a way of making it easier to understand

**Review by DrKimpatrick — APPROVED** [review]

**Review by collinewait — APPROVED** [review]

**Pull request merged**